### ISSUE TITLE
Hangs on connecting when starting docker-with-dotfiles

### ISSUE THREAD

**Issue by ecarlson94** [coder/coder#4052]
When deploying with the docker-with-dotfiles template, it seems the container can not download the required dependencies.

```
+ :

=
+ command -v curl
+ curl -fsSL --compressed https://coder.<domain>.com/bin/coder-linux-amd64 -o coder
curl: (22) The requested URL returned error: 503

+ status=22
+ echo error: failed to download coder agent

+ echo command returned: 22
+ echo Trying again in 30 seconds...
+ sleep 30
error: failed to download coder agent

command returned: 22
Trying again in 30 seconds...
```
I have navigated to the above URL and verified that I can download a file from there.

Here is the version of coder I am running in inside a docker container.

```
/home/coder $ /opt/coder version
Coder v0.8.14+e1afec6 Thu Sep  8 14:06:00 UTC 2022
https://github.com/coder/coder/commit/e1afec6db4cb72b4dbc8d955fab20dc5b369243c
```
 
I tried running the `curl` command on my local machine. I receive the same error.

**Comment by ericpaulsen** [comment]
hi @ecarlson94 - what is the docker image you're using with the template?

**Comment by ecarlson94** [comment]
Maybe that is the issue. I am using this image: https://hub.docker.com/r/cirrusci/flutter/. The error was occurring inside the `cirrusci/flutter` after the workspace was spun up.

**Comment by ericpaulsen** [comment]
@ecarlson94 I ran the template with your image, `cirrusci/flutter`, and am seeing a different behavior.

```
+ BINARY_URL=https://oss.ericpaulsen.io/bin/coder-linux-amd64
+ cd /tmp/coder.jR1tmf
+ :
+ status=
+ command -v curl
+ curl -fsSL --compressed https://oss.ericpaulsen.io/bin/coder-linux-amd64 -o coder
+ break
+ chmod +x coder
+ export CODER_AGENT_AUTH=token
+ export CODER_AGENT_URL=https://oss.ericpaulsen.io/
+ exec ./coder agent
```

the `503` error code suggests your coder server went down while the container was pulling the agent binary... can you try re-building the workspace and let me know if you continue seeing this behavior?

**Comment by ecarlson94** [comment]
Yes, I do have a proxy in front of coder. I'm using Nginx with the following server bloc for coder:
```
server {
    listen 443 ssl;
    listen [::]:443 ssl;

    server_name coder.<domain>.com;

    include /config/nginx/ssl.conf;

    client_max_body_size 0;

    location / {
        include /config/nginx/proxy.conf;
        include /config/nginx/resolver.conf;
        set $upstream_app coder;
        set $upstream_port 7080;
        set $upstream_proto http;
        proxy_pass $upstream_proto://$upstream_app:$upstream_port;
    }
}
```

I adapted the configuration from from [linuxserver.io's code-server template](https://github.com/linuxserver/reverse-proxy-confs/blob/master/code-server.subdomain.conf.sample). The webpage is accessible in a browser. And when I navigate to `https://coder.<domain>.com/bin/coder-linux-amd64` in a browser, a 41MB file is downloaded. But the curl command is failing with that 503 error.

**Comment by ericpaulsen** [comment]
@ecarlson94 - can you confirm for me whether `nginx` is also being run in a docker container, and whether it is on the same network as the `coder` container?

you can check a container's network via `docker inspect <container-id>`.

**Comment by ecarlson94** [comment]
They are both running in a user defined network called proxynet with no special settings. It is just the default new docker network. Specifically so that my containers can interact with one another using their container name. They both have this network configuration, with a few minor of course:
```
"Networks": {
    "proxynet": {
        "IPAMConfig": null,
        "Links": null,
        "Aliases": [
            "0f35984c0fad"
        ],
        "NetworkID": "2515bf8060900e4524df5a72c590749d3d5d212069783e8c5497504fb81d7e82",
        "EndpointID": "d3eaa54b3268526bcd55b93af5c9b14d4491f940f70927aee6806cc078dcd651",
        "Gateway": "172.21.0.1",
        "IPAddress": "172.21.0.25",
        "IPPrefixLen": 16,
        "IPv6Gateway": "",
        "GlobalIPv6Address": "",
        "GlobalIPv6PrefixLen": 0,
        "MacAddress": "02:42:ac:15:00:19",
        "DriverOpts": null
    }
}
```

**Comment by ericpaulsen** [comment]
thanks for confirming @ecarlson94 - can you confirm whether `http://coder:7080` is reachable from nginx?

**Comment by ecarlson94** [comment]
I can confirm. As I mentioned before, I can access https://coder.mydomain.com. And I can download the file at https://coder.mydomain.com/bin/coder-linux-amd64 which is proof that nginx is configured correctly and can access http://coder:7080.

**Comment by ecarlson94** [comment]
I can even download that file using that exact command using curl on windows. I even downloaded the file using that exact command in a terminal on my android phone (both on and off of my network). That command doesn't work on any of my linux desktops and containers.

**Comment by ecarlson94** [comment]
I have finally figured out the source of this. Over the past couple of weeks, I have noticed this issue cropping up across several of my self-hosted projects. The problem was that I recently moved. ACROSS the country. So, Cloudflare thought that all traffic coming from my network was suss and started issuing javascript challenges to almost all of my own resources. Ultimately silly since the target server is also on my network, but whatever.

Thanks for your help. Sorry to bother you.

**Comment by ericpaulsen** [comment]
No worries! Thanks for your persistence.
